src/cluster_slot_stats.c

```diff
@@ -113,6 +122,36 @@ static void addReplyOrderBy(client *c, int order_by, long limit, int desc) {
     addReplySortedSlotStats(c, slot_stats, limit);
 }
 
+/* Resets applicable slot statistics. */
+void clusterSlotStatReset(int slot) {
+    /* key-count is exempt, as it is queried separately through `countKeysInSlot()`. */
+    memset(&server.cluster->slot_stats[slot], 0, sizeof(slotStat));
+}
+
+void clusterSlotStatResetAll(void) {
+    if (server.cluster == NULL) return;
+
+    memset(server.cluster->slot_stats, 0, sizeof(server.cluster->slot_stats));
+}
+
+/* For cpu-usec accumulation, EXEC, EVAl and FCALL commands are skipped.
+ * This is due to their unique callstack, where the c->duration for
+ * EXEC, EVAL and FCALL already includes all of its nested commands.
+ * Meaning, the accumulation of cpu-usec for these wrapper commands
+ * would equate to repeating the same calculation twice.
```

Comment:
It doesn't include the CPU time spent within the lua function itself though, which seems odd. Since it is included in the command CPU.

Reply:
IMO, cross-slot scripts are not a use-case we need to care about, as long as they don't crash the aerver. I'm fine with metrics being completely screwed up for such scripts.

That said, i don't mind we pick option 1 or 2.

Reply:
Kyle and I discussed offline, and we recommend proceeding with 2 and we can get community input in RC1. It's easy to-revaluate this option later.